src/Algebra/HeytingField.agda

```diff
@@ -0,0 +1,99 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Local commutative rings and heyting fields
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.HeytingField where
+
+open import Algebra.Core using (Op₁; Op₂)
+open import Algebra.Structures using (IsCommutativeRing)
+open import Relation.Binary.Core using (Rel)
+open import Relation.Binary.Structures using (IsEquivalence)
+open import Level using (Level; _⊔_) renaming (suc to lsuc)
+open import Data.Product using (∃-syntax; _×_)
+
+import Relation.Binary.Apartness as A
+
+
+module Structures
+  {c ℓ₁ ℓ₂} {Carrier : Set c}
+  (_≈_ : Rel Carrier ℓ₁)
+  (_#_ : Rel Carrier ℓ₂)
+  (_+_ _*_ : Op₂ Carrier) (-_ : Op₁ Carrier) (0# 1# : Carrier)
+  where
+
+  open A.Structures hiding (¬#-equiv)
+
+  -- does this need to be public?
+  private
+    invertible : Carrier → Set _
+    invertible x = ∃[ 1/x ] (x * 1/x) ≈ 1#
+
+
+  record IsLocalCommutativeRing : Set (c ⊔ ℓ₁ ⊔ ℓ₂) where
+    field
+      isCommutativeRing : IsCommutativeRing _≈_ _+_ _*_ -_ 0# 1#
+      isApartness : IsApartness _≈_ _#_
+      #⇔invertible : ∀ {x y} → let x-y = x + (- y) in (x # y → invertible x-y) × (invertible x-y → x # y)
+
+    open IsCommutativeRing isCommutativeRing public
+    open IsApartness isApartness public
+
+    open import Relation.Nullary using (¬_)
```

Comment:
All import statements should go at the top of files :+1:

Reply:
Fixed.